src/js/base/translate-parse-tree.js

```diff
@@ -0,0 +1,1312 @@
+define("pyret-base/js/translate-parse-tree", [], function() {
+  
+  function translate(ast, fileName, constructors) {
+    // makeNode will be called on the node name, followed by the node args
+    let makeNode             = constructors.makeNode;
+    // opLookup should be a dictionary that
+    // maps bin-ops to strings and check-ops to (srcloc -> ast) functions.
+    let opLookup             = constructors.opLookup;
+    let makeLink             = constructors.makeLink;
+    let makeEmpty            = constructors.makeEmpty;
+    let makeString           = constructors.makeString;
+    let makeNumberFromString = constructors.makeNumberFromString;
+    let makeBoolean          = constructors.makeBoolean;
+    let makeNone             = constructors.makeNone; // RT.ffi.makeNone
+    let makeSome             = constructors.makeSome; // RT.ffi.makeSome
+    let getRecordFields      = constructors.getRecordFields; // getField(rec, 'fields')
+    // makeSrcloc: fileName, pos -> srcloc
+    let makeSrcloc         = constructors.makeSrcloc;
+    // combineSrcloc: fileName, pos, pos -> srcloc   -- beg of first to end of second
+    let combineSrcloc      = constructors.combineSrcloc;
+    // To improve error messages in an edge case. See parse-pyret.arr.
+    let detectAndComplainAboutOperatorWhitespace
```

Comment:
Not much to be done about this one :(

Reply:
Well, nothing to be done but give good names and document why it's there :-)